### PR TITLE
Fixes #23223 - Don't allow Dynflow to initialize in db:drop rake

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -276,7 +276,7 @@ class Application < Rails::Application
     end
 
     config.after_initialize do
-      init_dynflow
+      init_dynflow unless Foreman.in_rake?('db:drop')
```
Comment (ekohl) [inline config/application.rb:279]: Are there any rake tasks where it does need to be initialized?

**Comment by ekohl** [comment]
I've moved the redmine issue.

File: config/application.rb (hunk 1 above)
Comment (adamruzicka) [reply config/application.rb:279]: For a reason I don't recall now we load the Dynflow executor in `db:seed` and `db:migrate` (see[1]). Another place where we might be using it are foreman-tasks tests.

[1] - https://github.com/Dynflow/dynflow/blob/master/lib/dynflow/rails/configuration.rb#L40

Comment (adamruzicka) [reply config/application.rb:279]: cc @iNecas

Comment (iNecas) [reply config/application.rb:279]: The reason is that at the gtime of db:seed and db:migrate, the external executor might not be running, while we would need it for some seed of migration operations.

**Review by iNecas — APPROVED** [review]

**Comment by iNecas** [comment]
@adujicek

**Comment by mmoll** [comment]
test failures unrelated

**Comment by mmoll** [comment]
merged, díky @adamruzicka!

File: config/application.rb (hunk 1 above)
Comment (mbacovsky) [reply config/application.rb:279]: does this prevent us to use `db:purge`?

Comment (mbacovsky) [reply config/application.rb:279]: as per our discussion with @adamruzicka and some testing it does. However after some investigation I found out I won't be able to use `db:purge` anyway.

**Pull request merged**